### PR TITLE
Create TestController #950

### LINKED ISSUE #950: Create TestController
Mostly to allow calling of the `UI` instance during tests. 

Should help with #933.

### PR DESCRIPTION
**Pull request by lejolly**
Fixes #950

### CODE REVIEW

**Comment by acruis** [comment]
Do update the [testing documentation](https://github.com/HubTurbo/HubTurbo/blob/master/docs/testing.md) to mention the availability of the UI object while testing.

File: src/main/java/ui/TestController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    public static HashMap<String, String> getCommandLineArgs() {
+        return commandLineArgs;
+    }
+
+    // Test mode should only be run as a test task (Gradle / JUnit), as quit()
+    // leaves the JVM alive during test mode (which is cleaned up by Gradle).
+    // Manually feeding --test=true into the command line arguments will leave the JVM
+    // running after the HT window has been closed, and thus will require the
+    // process to be closed manually afterwards (Force Quit / End Process).
```
Comment (damithc) [inline src/main/java/ui/TestController.java]: Any reason why this is not a javadoc comment?

Comment (acruis) [inline src/main/java/ui/TestController.java]: It's supposed to apply to not just the `isTestMode` method, but also the other subsequent command line argument methods.

**Comment by lejolly** [comment]
ready for review.

**Comment by acruis** [comment]
Coveralls is acting up, but it's good to merge.

**Pull request merged**